src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -142,7 +142,7 @@ void SyncManager::do_make_logger()
     REALM_ASSERT(m_logger_ptr);
 }
 
-const std::shared_ptr<util::Logger>& SyncManager::get_logger() const
+std::shared_ptr<util::Logger> SyncManager::get_logger() const
```

Comment:
Returning a reference here exposed the field which was supposed to be guarded by the mutex.